scripts/xtensa-build-all.sh

```diff
@@ -113,171 +90,148 @@ set -e
 OLDPATH=$PATH
 WORKDIR="$pwd"
 
-# build platform
-for j in ${PLATFORMS[@]}
+# build platforms
+for platform in ${PLATFORMS[@]}
```

Comment:
It's a miracle this works considering PLATFORMS is not an array. This works by combining two bugs that cancel each other out:
- To list each element of the array it should be `for platform in "${PLATFORMS[@]}"` with double quotes
- but PLATFORMS is not an array but a string with whitespace

https://mywiki.wooledge.org/BashGuide/Arrays
Demo:
```diff
--- a/scripts/xtensa-build-all.sh
+++ b/scripts/xtensa-build-all.sh
@@ -90,6 +90,19 @@ else
        done
 fi
 
+# Here's an actual array
+declare -p SUPPORTED_PLATFORMS
+printf 'SPi=%s\n' "${SUPPORTED_PLATFORMS[@]}"
+
+# Not an array, just pretending
+declare -p PLATFORMS
+# yet bash expands this a string with whitespace, amazing
+printf 'Pi=%s\n' "${PLATFORMS[@]}"
+printf 'Pi=%s\n' "${PLATFORMS[@]}"
+
+# now something even weirder
+printf '?#?=%s\n' "${#PLATFORMS[@]}"
+
 # check target platform(s) have been passed in
 if [ ${#PLATFORMS[@]} -eq 0 ];
 then
@@ -97,6 +110,8 @@ then
        exit 1
 fi
 
+exit 0
+
 if [ "x$USE_PRIVATE_KEY" == "xyes" ]

```